internal/telemetry/client.go

```diff
@@ -31,7 +30,7 @@ import (
 // Client buffers and sends telemetry messages to Datadog (possibly through an
 // agent).
 type Client interface {
-	ProductStart(namespace Namespace, configuration []Configuration)
+	ProductChange(namespace Namespace, enabled bool, configuration []Configuration)
```

Comment:
Why are we only sending `app-product-change` messages now, instead of `app-started`? Are there any implications of that?

Reply:
We're still sending `app-started`, but only on the 1st call to `ProductChange`. There is actually a system test that asserts this behavior (and they still pass).

Reply:
I'd add that I'm not super fond of how the naming of the method is somewhat confusing here about this particular aspect of it, but I'm also not seeing any obviously better solution just yet...

Reply:
Gotcha, okay thanks.